frameworks/hdfs/tests/test_auth.py

```diff
@@ -87,28 +115,90 @@ def test_health_of_kerberized_hdfs():
-        assert stdout == config.TEST_CONTENT_SMALL
 
-    finally:
-        sdk_marathon.destroy_app(client_task_id)
 
+@pytest.mark.dcos_min_version('1.10')
+@sdk_utils.dcos_ee_only
+@pytest.mark.auth
+@pytest.mark.sanity
+def test_users_have_appropriate_permissions(kerberized_hdfs_client):
+    # "hdfs" is a superuser
+    sdk_auth.kinit(kerberized_hdfs_client, keytab=config.KEYTAB, principal=config.CLIENT_PRINCIPALS["hdfs"])
+
+    for user, _ in config.CLIENT_PRINCIPALS.items():
+        log.info("Creating directory for user: {}".format(user))
+        make_user_directory_cmd = config.hdfs_command("mkdir -p /users/{user}".format(user=user))
+        sdk_tasks.task_exec(kerberized_hdfs_client, make_user_directory_cmd)
+
+        change_ownership_cmd = config.hdfs_command("chown {user}:users /users/{user}".format(user=user))
+        sdk_tasks.task_exec(kerberized_hdfs_client, change_ownership_cmd)
+
+        change_permissions_cmd = config.hdfs_command("chmod 700 /users/{user}".format(user=user))
+        sdk_tasks.task_exec(kerberized_hdfs_client, change_permissions_cmd)
+
+    # alice has read/write access to her directory
+    sdk_auth.kdestroy(kerberized_hdfs_client)
+    sdk_auth.kinit(kerberized_hdfs_client, keytab=config.KEYTAB, principal=config.CLIENT_PRINCIPALS["alice"])
+    write_access_cmd = "/bin/bash -c \"{}\"".format(config.hdfs_write_command(
+        config.TEST_CONTENT_SMALL,
+        "/users/alice/{}".format(config.TEST_FILE_1_NAME)))
+    log.info("Alice can write: {}".format(write_access_cmd))
+    rc, stdout, _ = sdk_tasks.task_exec(kerberized_hdfs_client, write_access_cmd)
+    assert stdout == '' and rc == 0
+
+    read_access_cmd = config.hdfs_read_command("/users/alice/{}".format(config.TEST_FILE_1_NAME))
+    log.info("Alice can read: {}".format(read_access_cmd))
+    _, stdout, _ = sdk_tasks.task_exec(kerberized_hdfs_client, read_access_cmd)
+    assert stdout == config.TEST_CONTENT_SMALL
+
+    ls_cmd = config.hdfs_command("ls /users/alice")
+    _, stdout, _ = sdk_tasks.task_exec(kerberized_hdfs_client, ls_cmd)
+    assert "/users/alice/{}".format(config.TEST_FILE_1_NAME) in stdout
+
+    # bob doesn't have read/write access to alice's directory
+    sdk_auth.kdestroy(kerberized_hdfs_client)
+    sdk_auth.kinit(kerberized_hdfs_client, keytab=config.KEYTAB, principal=config.CLIENT_PRINCIPALS["bob"])
+
+    write_access_cmd = "/bin/bash -c \"{}\"".format(config.hdfs_write_command(
+        config.TEST_CONTENT_SMALL,
+        "/users/alice/{}".format(config.TEST_FILE_1_NAME)))
+    log.info("Bob tries to wrtie to alice's directory: {}".format(write_access_cmd))
+    _, _, stderr = sdk_tasks.task_exec(kerberized_hdfs_client, write_access_cmd)
+    log.info("Bob can't write to alice's directory: {}".format(write_access_cmd))
+    assert "put: Permission denied: user=bob" in stderr
+
+    read_access_cmd = config.hdfs_read_command("/users/alice/{}".format(config.TEST_FILE_1_NAME))
```

Comment:
This is the same read command as above.

Reply:
Removed both latter `write_access_cmd` and `read_access_cmd`.